pywren/future.py

```diff
@@ -73,8 +73,16 @@ def _set_state(self, new_state):
         ## FIXME add state machine
         self._state = new_state
 
-    def cancel(self):
-        raise NotImplementedError("Cannot cancel dispatched jobs")
+    def cancel(self, storage_handler=None):
+        # FIRST This is annoying; how will it have the custom access
+        # to the storage handler information
+
+        if storage_handler is None:
+            storage_config = wrenconfig.extract_storage_config(wrenconfig.default())
+            storage_handler = storage.Storage(storage_config)
+
+        storage_handler.put_cancelled(self.callset_id,
+                                      self.call_id, "foo")
```

Comment:
Can we not hard code "foo" in the code ? Create a constant ? Also will this field be used for anything else in the future ? If so can we create a map that says "STATUS_CANCELED: 1" and write the value from that map here ?

Reply:
Since the only information we need to cancel a job is the _existence_ of a key, I don't think the value is meaningful. We'd never need to change it. In this case, I don't think hardcoding a value is bad; I feel that creating such a map would create unnecessary clutter. I will change the constant from "foo" to "CANCEL", however.